examples/agriculture-demo.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (C) 2024 Robotec.AI
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import argparse
+import subprocess
+import threading
+import time
+
+import rclpy
+
+from rai.node import RaiStateBasedLlmNode, describe_ros_image
+from rai.tools.ros.native import (
+    GetCameraImage,
+    GetMsgFromTopic,
+    Ros2GenericServiceCaller,
+    Ros2GetRobotInterfaces,
+    Ros2ShowMsgInterfaceTool,
+)
+from rai.tools.time import WaitForSecondsTool
+
+
+def mock_behavior_tree(tractor_number: int):
+    """
+    This is a mock of behavior tree that simulates scenario where the tractor stops
+    due to an unexpected situation and calls the RaiNode to decide what to do.
+    """
+
+    while True:
+        output = subprocess.check_output(
+            [
+                "ros2",
+                "service",
+                "call",
+                f"/tractor{tractor_number}/current_state",
+                "std_srvs/srv/Trigger",
+                "{}",
+            ]
+        )
+        print(output.decode("utf-8"))
+        if "STOPPED" in output.decode("utf-8"):
+            print("The tractor has stopped. Calling RaiNode to decide what to do.")
+            action_output = subprocess.check_output(
+                [
+                    "ros2",
+                    "action",
+                    "send_goal",
+                    "-f",
+                    "/perform_task",
+                    "rai_interfaces/action/Task",
+                    "{priority: 10, description: '', task: 'Anomaly detected. Please decide what to do.'}",
+                ]
+            )
+            print(action_output.decode("utf-8"))
+        time.sleep(5)
```

Comment:
How about making it a ros node with a timer instead of using subprocess?

Reply:
[7febe0d](https://github.com/RobotecAI/rai/pull/285/commits/7febe0dd723b501fbb30d639680661e4565a50b9)